chapters/extend/driver.rst

```diff
@@ -1,16 +1,153 @@
-Guideline When Adding New Driver
-================================
+Development Guide: Add new Drivers
+====================================
 
-You come to this page and you realize the existing executor/driver logic may not suit you need, so you want to write a new driver and add to Jina.
+.. meta::
+   :description: Development Guide: Add new Drivers
+   :keywords: Jina, driver
 
-Or *do* you?
+.. note:: This guide assumes you have a basic understanding of Jina, if you haven't, please check out `Jina 101 <https://101.jina.ai>`_ first.
+.. note:: Before reading this guide, you might want to read `Development Guide: Add new Executors <../executor.rst>`_ first.
 
-In most of the case, you don't.
+.. contents:: Table of Contents
+    :depth: 2
 
-If you do ... Docs needs
+Motivation
+^^^^^^^^^^^
 
+As you might have already learned from `Jina 101 <https://101.jina.ai>`_,
+a :term:`Driver` interprets incoming messages into :term:`Document` and extracts required fields for an :term:`Executor`.
 
+Jina already created `several Drivers <https://docs.jina.ai/chapters/all_driver/>`_ Drivers inside Core,
```

Comment:
this comes out as driver Drivers inside Core?

Reply:
I'll rephrase to Jina Core :)

Reply:
@FionnD I don't understand your comment, can you elaborate a bit on this?

Reply:
See my slack message